### ISSUE TITLE
Vue Production Mode: DOMException: Failed to execute 'appendChild' on 'Node': This node type does not support this method.

### ISSUE THREAD

**Issue by haelbichalex** [DivanteLtd/vue-storefront#3669]
Related issues: #3190, #2216, #1642 
When reloading a category page, I get the following error in my console: 
```
DOMException: Failed to execute 'appendChild' on 'Node': This node type does not support this method.
```

This is not really a bug for the releases listed below, since I can't reproduce the error there. I've seen this issue occurred a couple of times before and I want to find out what's happening there. Note that I'm already running the latest VSF version 1.10.3. Also this issue only occurs when the frontend is in production mode, in dev mode everything works.
The most strange thing is, I could "fix" the error by wrapping my category content inside another `<div>` element in the `src/themes/guf/pages/Category.vue` component:
```vue
<template>
  <div id="category">
    <!-- This div is needed to fix #1681. -->
    <div>
    ...
    content
    ...
    </div>
  </div>
</template>
```

Maybe someone can give me a hint of what could be happening here and why I get the error.

## Can you handle fixing this bug by yourself?

- [ ] YES
- [x] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: Chrome 77
- OS: Mac OS
- Node: v10.16.3

**Comment by haelbichalex** [comment]
Also interesting is this PR: #2220

**Comment by pkarw** [comment]
I've changed to `question` - I'm not sure if it's a bug - we need to confirm

**Comment by NicolasHofmairT23** [comment]
Hi, @pkarw We are still challenging this issue. Could you confirm the issue and could provide some information to this topic?

**Comment by pkarw** [comment]
@NicolasHofmairT23 I can't confirm this issue. We didn't get any other report on it + our QA was not able to reproduce.

You've got it on 1.10 or 1.11?

@alinadivante can we take a second look at this?

**Comment by alinadivante** [comment]
@pkarw I tried to reproduce this error on _demo_, _next_ and _test_, but I  also can't confirm it

**Comment by janfeldmann** [comment]
Hey, we can reproduce this issue. We had the following structure inside 
/layouts/Default.vue

`<main>
        <slot />
       <footer-contact />
</main>`

Moving the <footer-contact /> Element out of the <main> Container resolved the issue.

By debugging via Chrome DevTools I was able to see that the node type that the appendChild Method was used on was "text".

**Comment by janfeldmann** [comment]
@haelbichalex 
Yeah :D copy 'n paste got me there
I updated my comment above

**Comment by andrzejewsky** [comment]
@janfeldmann that sounds like trail, we'll try again to solve this, one question - which version of vsf did you use?

**Comment by janfeldmann** [comment]
@andrzejewsky we are using version 1.10.2

**Comment by haelbichalex** [comment]
What also might be of interest: We get this error only when loading category pages directly via the url. When first loading the homepage for example, then navigate via the menu to the category, everything's working.

**Comment by haelbichalex** [comment]
I found something, which I also could reproduce on the default theme without any changes using v1.10.4:
When loading a category page "a couple of times", sometimes it renders everything above the `<slot />` component in [Default.vue](https://github.com/DivanteLtd/vue-storefront/blob/e44e6fdfe7ff98af32c852905db67ef3f47e9c33/src/themes/default/layouts/Default.vue) and moves the `Category` component to the bottom of all components:
![image](https://user-images.githubusercontent.com/15052352/68840897-e181f000-06c3-11ea-9f18-6695d2309686.png)

This then might cause the issue which is also visible in development mode:
![image](https://user-images.githubusercontent.com/15052352/68840972-05ddcc80-06c4-11ea-8874-78b37e3472f7.png)

Can you reproduce this?

**Comment by pkarw** [comment]
@patzick @andrzejewsky it looks like an issue we already solved somewhere between 1.10.2 and 1.10.4 can you guys reproduce it?

**Comment by pkarw** [comment]
I’m not sure what can cause this issue @haelbichalex described. I’m concern about the lazy loading sub components we’re using to lazy load the layouts (themes/default/App.vue - https://github.com/DivanteLtd/vue-storefront/blob/master/src/themes/default/App.vue)

Some ideas for the investigation:
- putting the <slot /> into <div></div>
- trying to simplify the theme by removing the dynamic layouts layer - I mean we should have the  layouts/Default.vue merged to App.vue and use this single component file for the layout; of course it’s not preferable solution as we’re removing the dynamic layout feature by this move; it’s rather just for an investigation
- investigate the url/router/beforeEach.ts vs client-entry.ts hydration hook in order to make sure the hydration takes place only after the url dispatcher resolved the right route and not before (bc. otherwise the wrong component, assigned to the route could be loaded) - our recent fix to similar issue done by @patzick based on removing this race condition already (1.10.3)
